shapely/geometry/polygon.py

```diff
@@ -24,35 +24,39 @@ def _unpickle_linearring(wkb):
 
 class LinearRing(LineString):
     """
-    A closed one-dimensional feature comprising one or more line segments
+    A closed one-dimensional feature comprising one or more line segments.
 
     A LinearRing that crosses itself or touches itself at a single point is
     invalid and operations on it may fail.
+
+    Parameters
+    ----------
+    coordinates : sequence
+        A sequence of (x, y [,z]) numeric coordinate pairs or triples.
+        Also can be a sequence of Point objects.
+
+    Notes
+    -----
+    Rings are implicitly closed. There is no need to specify a final
+    coordinate pair identical to the first.
+
+    Examples
+    --------
+    Construct a square ring.
+
+    >>> ring = LinearRing( ((0, 0), (0, 1), (1 ,1 ), (1 , 0)) )
+    >>> ring.is_closed
+    True
+    >>> list(ring.coords)
+    [(0.0, 0.0), (0.0, 1.0), (1.0, 1.0), (1.0, 0.0), (0.0, 0.0)]
+    >>> ring.length
+    4.0
```

Comment:
shouldn't lengths always be whole integers?

Reply:
This is a bit confusing, but so it's the number of line segments, but the actual length of the line, and that is correctly a float.

Reply:
I don't think there is actually a direct property on the class that gives this. There is the new top-level `shapely.get_num_coordinates` function, or one could do `len(ring.coords)`

Reply:
🤦 `ring.length` != `len(ring)`; sorry - I wasn't reading this carefully enough!